R/histogram.R

```diff
@@ -1,17 +1,86 @@
 #' Create a histogram with the magma color scheme
 #'
 #' @param df A dataframe containing the variables for plotting
-#' @param x Column name of the variable to be plotted on the x-axis
-#' @param y An aggregation function in string version to be plotted on the y-axis.
-#'            The supported aggregation operations are: ['average', 'count',
-#'            'distinct', 'max', 'mean', 'median', 'min', 'missing', 'product',
-#'            'q1', 'q3', 'ci0', 'ci1', 'stderr', 'stdev', 'stdevp', 'sum', 
-#'            'valid', 'values', 'variance', 'variancep']
+#' @param x Column name of the numeric variable to be plotted on the x-axis
+#' @param y The quoted aggregation function to be plotted on the y-axis
+#'            and to be used for the fill color.
+#'            The input needs to be in the form of `"..name.."`,
+#'            where name can be values from the following list,
+#'            with explanations in the parenthesis:
+#'            [count (number of points in bin),
+#'             density (density of points in bin, scaled to integrate to 1),
+#'             ncount (count, scaled to maximum of 1),
+#'             ndensity (density, scaled to maximum of 1),
+#'             width (widths of bins)]
 #' @return A ggplot object.
 #' @export
 #'
 #' @examples
-#' histogram(mtcars, vs, 'min(mpg)')
-#' histogram(mtcars, vs, 'count()')
+#' histogram(mtcars, vs, "..count..")
+#' histogram(mtcars, vs, "..ndensity..')
+
+library(ggplot2)
+library(dplyr)
+library(rlang)
+library(viridis)
+
 histogram <- function(df, x, y) {
-}
\ No newline at end of file
+  aggregation_functions = c("..count..", "..density..",
+                            "..ncount..", "..ndensity..",
+                            "..width..")
+  
+  # check to ensure df is assigned a dataframe
+  if (!is.data.frame(df)) {
+    stop("Data assigned to 'df' must be a dataframe.")
+  }
+  
+  # check to ensure x is not in quotes
+  if (
+    typeof(substitute(x)) !=
+    typeof(as.list(quote(symbol))[[1]])
+  ) {
+    stop("Column name assigned to 'x' must not be in quotes.")
+  }
+  
+  # check to ensure x is present in the df
+  if (!as.character(ggplot2::vars({{ x }})[[1]])[2] %in% colnames(df)) {
+    stop("Column assigned to 'x' is not found in dataframe.")
+  }
+  
+  # check to ensure column x is numeric 
+  if (!is.numeric(dplyr::pull(df, {{x}}))) {
+    stop("Column assigned to 'x' is not numeric.")
+  }
+  
+  # check to ensure y is in quotes
+  if (
+    typeof(substitute(y)) == typeof(as.list(quote(symbol))[[1]])
+  ) {
+    stop("The aggregation function assigned to 'y' must be in quotes.")
+  }
+
+  # check to ensure y is in the correct format and has the supported value
+  if (!startsWith(y, "..") || !endsWith(y, "..") ) {
+    stop("Value assigned to 'y' should start with '..' and end with '..'.")
```

Comment:
Just curious Irene, isn't this redundant as you are already checking to see if the inout for y exists in the `aggregation functions` list?

Reply:
Oh yeah you are right! Thank you for catching this!